sdks/java/testing/test-utils/src/main/java/org/apache/beam/sdk/testutils/jmh/Main.java

```diff
@@ -0,0 +1,184 @@
+      org.openjdk.jmh.Main.main(args);
+      return;
+    }
+
+    final Runner runner = new Runner(opts);
+    final Collection<RunResult> results = runner.run();
+
+    final Collection<DataPoint> dataPoints =
+        results.stream()
+            .filter(r -> r.getParams().getMode() != SingleShotTime)
+            .map(r -> dataPoint(influxDB.measurement, r))
+            .collect(toList());
+
+    InfluxDBPublisher.publish(influxDB, dataPoints);
+  }
+
+  private static boolean isSingleShotTimeOnly(Collection<Mode> modes) {
+    return !modes.isEmpty() && modes.stream().allMatch(SingleShotTime::equals);
+  }
+
+  private static DataPoint dataPoint(String baseMeasurement, RunResult run) {
+    final BenchmarkParams params = run.getParams();
+    final Result<?> result = run.getPrimaryResult();
+
+    final long startTimeMs =
+        metaDataStream(run).mapToLong(BenchmarkResultMetaData::getStartTime).min().getAsLong();
+    final long stopTimeMs =
+        metaDataStream(run).mapToLong(BenchmarkResultMetaData::getStopTime).max().getAsLong();
+
+    final String measurement =
+        String.format("%s_%s", baseMeasurement, params.getMode().shortLabel());
+
+    final Map<String, String> tags = new HashMap<>();
+    tags.put("benchmark", params.getBenchmark());
+    tags.put("scoreUnit", result.getScoreUnit());
+    // add params of parameterized benchmarks as tags
+    tags.putAll(params.getParamsKeys().stream().collect(toMap(identity(), params::getParam)));
+
+    final Map<String, Number> fields = new HashMap<>();
+    fields.put("score", result.getScore());
+    fields.put("scoreMean", result.getStatistics().getMean());
+    fields.put("scoreMedian", result.getStatistics().getPercentile(0.5));
+    if (!Double.isNaN(result.getScoreError())) {
+      fields.put("scoreError", result.getScoreError());
+    }
+    fields.put("sampleCount", result.getSampleCount());
+    fields.put("durationMs", stopTimeMs - startTimeMs);
+
+    return InfluxDBPublisher.dataPoint(
+        measurement, tags, fields, startTimeMs, TimeUnit.MILLISECONDS);
+  }
+
+  private static Stream<BenchmarkResultMetaData> metaDataStream(RunResult runResult) {
+    return runResult.getBenchmarkResults().stream()
+        .map(BenchmarkResult::getMetadata)
+        .filter(Objects::nonNull);
+  }
+
+  /** Construct InfluxDB settings from environment variables to not mess with JMH args. */
+  private static @Nullable InfluxDBSettings influxDBSettings() {
+    String host = System.getenv(INFLUXDB_HOST);
+    String database = System.getenv(INFLUXDB_DATABASE);
+    String measurement = System.getenv(INFLUXDB_BASE_MEASUREMENT);
+    if (measurement == null || database == null) {
+      return null;
+    }
+
+    InfluxDBSettings.Builder builder = InfluxDBSettings.builder();
+    if (host != null) {
+      builder.withHost(host); // default to localhost otherwise
+    }
+    return builder.withDatabase(database).withMeasurement(measurement).get();
```

Comment:
What is the purpose of specifying the measurement here?

Reply:
Thx for catching, fixed 👍